### ISSUE TITLE
Multi-part sankey?

### ISSUE THREAD

**Issue by emeeks** [d3/d3-sankey#11]
I implemented a multi-part sankey that allows for the individual links to be split into categorized pieces. Probably be nice to fold that into the base implementation so people have access to the functionality.

https://github.com/emeeks/d3.sankey-multipart/

**Comment by mbostock** [comment]
Is this different than giving as input a graph where there are multiple links between the same pair of nodes? For example:

```js
sankey({
  nodes:[
    {name:"A"},
    {name:"B"},
    {name:"C"}
  ],
  links:[
    {source:0, target:1, value:100, color: "red"},
    {source:0, target:1, value:80, color: "green"},
    {source:0, target:1, value:50, color: "blue"},
    {source:0, target:2, value:50, color: "red"},
    {source:0, target:2, value:40, color: "green"},
    {source:0, target:2, value:20, color: "blue"}
  ]
})
```

Which might look like:

<img width="960" alt="screen shot 2017-06-12 at 11 30 01 am" src="https://user-images.githubusercontent.com/230541/27048949-909815d6-4f62-11e7-954f-099c91fce628.png">

The thing I could imagine that’s missing is a constraint that the links are drawn in a certain order.

**Comment by emeeks** [comment]
Pretty sure I made it because I thought the sankey layout didn't support parallel edges. It's been a while, if there was another reason and I remember, I'll post it, otherwise yeah, just the ability to determine draw order would be useful.

**Comment by mbostock** [comment]
It occurs to me that the requested feature, then—control over the order of links—was already implemented in #19. Hooray.
